### ISSUE TITLE
[React Delivery form] Refactor the time slot selection modal

### ISSUE THREAD

**Issue by Paul-Eraman-CoopCycle** [coopcycle/coopcycle-web#4320]
As a store owner or dispatcher when making an order (ex: https://demo.coopcycle.org/admin/stores/87/deliveries/new) 
 Right now we have one dropdown menu with the day and time together. This makes a long and difficult to read list.

I would like to change this interface into something with clickable options for first the day, and then the time:
![image](https://github.com/coopcycle/coopcycle-web/assets/77277854/555e1898-3c4d-4987-8a31-c47bb891932f)

Additional information:

https://coopcycle.slack.com/archives/C4UT4DGQ3/p1728058481085709

**Comment by Atala** [comment]
i would add as linked issue to find a better date (and time picker) before we do this issue, so we start replacing our current antd datetime picker which i find awful (but maybe you like it tell me)
- current https://ant.design/components/date-picker
- possibility 1, you can fill with tab tab in 10 seconds https://codesandbox.io/p/sandbox/white-breeze-rer8g2?file=%2Fsrc%2FApp.js (from https://react-spectrum.adobe.com/react-aria/useDatePicker.html)

**Comment by Paul-Eraman-CoopCycle** [comment]
Current one is awful, if you have a good one id be super happy if it changed asap

**Comment by Atala** [comment]
the one i proposed above seems good to me
but you/we can find a lot of them, just google "react datetime picker" and you will get options

**Comment by tcmjlnt** [comment]
yes, that date picker option seems totally adequate to me as well

**Comment by Atala** [comment]
created issue https://github.com/coopcycle/coopcycle-web/issues/4720

**Comment by parappe-coopcyclette** [comment]
Hey, was looking for this option! I see we're not a lot to use this for local commerce.  As we're a new cooperative, we would like to setup our new clients with Coopcycle directly and the date selector is blocking us. 
We know some of our regular client will make a lot of mistakes when encoding theyr deliveries. (2weeks, from 7am to 6pm 15min frame slot do quite a lot of possibilities). 
So we are using a shared excel for the moment but as we're trying to expand to other local commerce, this will not hold for long (takes a lot of time and so at the end we do not use Coopcycle software).
I don't know how the enhancment feature are prioritized and how we can help (a long time I didn't code but I may search for the date picker).

The date and time picker proposed seems good to me, I supposed the dropdown menu was helpfull to have proposition of date and time only based on the timeslot defined by the coop. (and so no validation needed).

**Comment by Atala** [comment]
hello,

we know it is the pain we gonna fix it but i can not promise a date. you have two options for me
- use timeslot but with big delivery windows so there is not too much choices (you can do 2-hour delivery window like here) : https://demo.coopcycle.org/admin/settings/time-slots/28
- do not use timeslots and let the store owner choose whatever they like
